jmh-generator-annprocess/src/main/java/org/openjdk/jmh/generators/annotations/APGeneratorDestinaton.java

```diff
@@ -53,8 +54,12 @@ public InputStream getResource(String resourcePath) throws IOException {
     }
 
     @Override
-    public Writer newClass(String className) throws IOException {
-        return processingEnv.getFiler().createSourceFile(className).openWriter();
+    public Writer newClass(String className, String originatingClassName) throws IOException {
+        if(originatingClassName != null) {
+            return processingEnv.getFiler().createSourceFile(className).openWriter();
+        } else {
+            return processingEnv.getFiler().createSourceFile(className, processingEnv.getElementUtils().getTypeElement(originatingClassName)).openWriter();
+        }
```

Comment:
Um, this branch is the wrong way. `originatingClassName` is guaranteed to be `null` at `else` branch.

```suggestion
    @Override
    public Writer newClass(String className, String originatingClassName) throws IOException {
        Filer filer = processingEnv.getFiler();
        if (originatingClassName != null) {
            TypeElement originatingType = processingEnv.getElementUtils().getTypeElement(originatingClassName);
            return filer.createSourceFile(className, originatingType).openWriter();
        } else {
            return filer.createSourceFile(className).openWriter();
        }
    }
```